### PR TITLE
[BACKLOG-2034]Disallow elements with empty names

### PR DESCRIPTION
**Pull request by hudak**
http://jira.pentaho.com/browse/BACKLOG-2034

### CODE REVIEW

**Comment by hudak** [comment]
@mdamour1976 @e-cuellar

File: test-src/org/pentaho/metastore/test/MetaStoreFactoryTest.java [hunk 1]
```diff
@@ -296,6 +297,29 @@ private Cube generateCube() {
...
+      element.setName( null );
+      factory.saveElement( element );
+      fail( "Saved illegal element (name == null)" );
+    } catch ( MetaStoreException e ) {
+      assertNotNull( e );
+    }
+
+    try {
+      MyElement element = new MyElement();
+      element.setName( "" );
```
Comment (dkincade) [inline test-src/org/pentaho/metastore/test/MetaStoreFactoryTest.java]: What about the test case where the element name is "   " (to validate the name is being trimmed). We may also need a test case checking for duplicates where the names are the same except the whitespace...

File: test-src/org/pentaho/metastore/test/MetaStoreFactoryTest.java [hunk 2]
```diff
@@ -296,6 +297,39 @@ private Cube generateCube() {
...
+    try {
+      element.setName( "" );
+      factory.saveElement( element );
+      fail( "Saved illegal element (name.isEmpty())" );
+    } catch ( MetaStoreException e ) {
+      assertNotNull( e );
+    }
+
+    try {
+      element.setName( " " );
```
Comment (hudak) [inline test-src/org/pentaho/metastore/test/MetaStoreFactoryTest.java]: @dkincade 
Updated the test to check for blank, non-empty names.
Any name sanitation should probably be done either a level up in the UI Controller or a level down in the specific metastore implementation.
Since the name is pulled off of the element with reflection, changing it might suggest setting it back to element with reflection. Element might be read only or reference elsewhere? Who knows. Gets messy

**Comment by dkincade** [comment]
@rfellows @bytekast
